src/iosMain/kotlin/com/airthings/lib/logging/platform/PlatformFileInputOutputImpl.kt

```diff
@@ -163,13 +158,15 @@ internal actual class PlatformFileInputOutputImpl : PlatformFileInputOutput {
         )
     }
 
-    private fun NSERROR_CPOINTER.sizeImpl(path: String): Long = NSFileManager.defaultManager
-        .attributesOfFileSystemForPath(path, this)
-        ?.get(NSFileSize)
-        ?.toString()
-        ?.toLongOrNull()
-        ?.coerceAtLeast(0L)
-        ?: throw IllegalArgumentException("Cannot get size of file: $path")
+    private fun sizeImpl(path: String): Long {
+        val fileHandle = NSFileHandle.fileHandleForReadingAtPath(path)
+            ?: throw IllegalArgumentException("Cannot open file for reading: $path")
+        return try {
+            fileHandle.seekToEndOfFile().toLong()
+        } finally {
+            fileHandle.closeFile()
+        }
+    }
```

Comment:
Not fond of opening a file just to get its size, this data is stored in the disk's filesystem and can be obtained quicker and without wasting CPU cycles for opening, seeking into, and closing the file.

How about this untested code:

```swift
val fileAttributes = NSFileManager.defaultManager.attributesOfItemAtPath(path, null)
val fileSize = fileAttributes?.get(/* FileAttributeKey.size */ "size")
```

I searched for a value compatible with Swift's `FileAttributeKey.size`, but couldn't find it. Maybe you can dump the Map's value to the console and see which key is the one for size?

Reply:
Sure, I'll take a look!

Reply:
@airxnoor Please take a look :)

Reply:
Sees like this can throw now (tostring -> toLong if it's null), fixing.

Reply:
Should probably return -1 if the file is unreadable or an error occurs, wdyt?

Reply:
The old implementation returned 0 if there was an error, which I carried over into the new. Still OK?

Reply:
A good question... I don't mind returning 0 still.